python/lsst/ts/m2gui/controltab/tab_rigid_body_pos.py

```diff
@@ -33,7 +37,326 @@ class TabRigidBodyPos(TabDefault):
+
+        # Absolute target position
+        self._target_position_absolute = self._set_target_position()
+
+        # Buttons in this table
+        self._buttons = {
+            "save_position": set_button("Save Position", self._callback_save_position),
+            "go_to_home": set_button("Go to Home", self._callback_go_to_home),
+            "set_home": set_button("Set Home", self._callback_set_home),
+            "jog": set_button("Jog", self._callback_jog),
+            "clear_values_relative": set_button(
+                "Clear Values",
+                self._callback_clear_values,
+                self._target_position_relative,
+            ),
+            "go_to_position": set_button(
+                "Go to Position", self._callback_go_to_position
+            ),
+            "clear_values_absolute": set_button(
+                "Clear Values",
+                self._callback_clear_values,
+                self._target_position_absolute,
+            ),
+        }
+
+        # Rigid body position
+        self._position = self._set_position()
+
+        # Internal layout
+        self.widget().setLayout(self._create_layout())
+
+        self._set_signal_position(self.model.utility_monitor.signal_position)
+
+    def _set_target_position(self):
+        """Set the target position of rigid body.
+
+        Returns
+        -------
+        position : `dict`
+            Target position of the rigid body. The key is the axis. The value
+            is the `PySide2.QtWidgets.QDoubleSpinBox`.
+        """
+
+        num_digit_after_decimal = (
+            self.model.utility_monitor.NUM_DIGIT_AFTER_DECIMAL_DISPLACEMENT
+        )
+
+        position = dict()
+        for axis in self._get_axes():
+            double_spin_box = QDoubleSpinBox()
+            double_spin_box.setDecimals(num_digit_after_decimal)
+            double_spin_box.setSingleStep(get_tol(num_digit_after_decimal))
+
+            if axis.startswith("r"):
+                double_spin_box.setRange(
+                    -self.MAX_ROTATION_IN_URAD, self.MAX_ROTATION_IN_URAD
+                )
+            else:
+                double_spin_box.setRange(
+                    -self.MAX_DISTANCE_IN_MM, self.MAX_DISTANCE_IN_MM
+                )
+
+            position[axis] = double_spin_box
+
+        return position
+
+    def _get_axes(self):
+        """Get the axes of position.
+
+        Returns
+        -------
+        `list`
+            Axes of the position.
+        """
+
+        return ["x", "y", "z", "rx", "ry", "rz"]
+
+    @Slot()
+    def _callback_save_position(self):
+        """Callback of the save-position button. This will save the current
+        position to a file."""
+        self.model.save_position()
+
+    @Slot()
+    def _callback_go_to_home(self):
+        """Callback of the go-to-home button. This will move the M2 to the
+        home position."""
+        self.model.go_to_position(0, 0, 0, 0, 0, 0)
```

Comment:
What will happen if that raise an exception? See _go_to_position. Probably a method _go_to_position with error handling (e.g. display a dialog saying you cannot go) would be  solution. See callback_go_to_position. You might as well end up with decorator - see https://medium.com/swlh/handling-exceptions-in-python-a-cleaner-way-using-decorators-fae22aa0abec for idea how.

Reply:
The error handling will be implemented in DM-34861. Thanks!